packages/edit-site/src/components/post-list/index.js

```diff
@@ -348,9 +348,16 @@ export default function PostList( { postType } ) {
 		context: 'list',
 	} );
 	const editAction = useEditPostAction();
+	const siteSettings = useSelect( ( select ) => {
+		return select( coreStore ).getEntityRecord( 'root', 'site' );
+	} );
+	const pageOnFront = siteSettings?.page_on_front;
+	const pageForPosts = siteSettings?.page_for_posts;
 	const actions = useMemo(
 		() => [ editAction, ...postTypeActions ],
-		[ postTypeActions, editAction ]
+		// The "Set as homepage" action eligibility depends on the pageOnFront and pageForPosts settings.
```

Comment:
I don't get this change..

Reply:
I added `pageOnFront` and `pageForPosts` as dependencies as the list of actions needs to re-render when either of these changes, otherwise the "Set as homepage" action will show incorrectly in the list of available actions for the page the user has just set as the homepage (it will show correctly after a page reload). I'm not sure how best to handle this!